### PR TITLE
Disable ForceNew for project sink unique_writer_identity field

### PR DESCRIPTION
**Pull request by pengq-google**
fixes https://github.com/hashicorp/terraform-provider-google/issues/15266

If this PR is for Terraform, I acknowledge that I have:

- [X] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [X] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/third_party/terraform/tests) (for handwritten resources or update tests).
- [X] [Generated Terraform providers](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/docs/getting-started/generate-providers.md), and ran [`make test` and `make lint`](https://googlecloudplatform.github.io/magic-modules/docs/getting-started/run-provider-tests/#run-unit-tests) in the generated providers to ensure it passes unit and linter tests.
- [X] [Ran](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/develop/run-tests.md) relevant acceptance tests using my own Google Cloud project and credentials (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [X] Read [Write release notes](https://googlecloudplatform.github.io/magic-modules/contribute/release-notes/) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:bug
Disable ForceNew for project sink unique_writer_identity field, so that sink can be updated in place. 
```

### CODE REVIEW

**Review by shuyama1 — COMMENTED** [review]
Thanks for making the change!

A few thoughts on the change though:
1. yes, removing the `forcenew` flag should support in-place update in Terraform (instead of deleting and recreating the resource) if the API supports it
2. However, I think the issue in https://github.com/hashicorp/terraform-provider-google/issues/15266 is more than just supporting in-place update for the field. 

	The issue seems like that if users specify `unique_writer_identity=false` explicitly or simply omit the field in their configs (they both mean the same as we have [terraform side default `false`](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/mmv1/third_party/terraform/services/logging/resource_logging_project_sink.go#L39) for the field, so terraform will send `unique_writer_identity=false` to the API in both cases) but the API returns `unique_writer_identity=true` (that's what writes back to terraform state) which causes a drift in Terraform. Running `terraform apply` again will try to update `unique_writer_identity` from `true` (what's in the terraform state after creation) to `false` to match the config.

	I happened to find the config you used to reproduce the issue internally. Could you try removing `unique_writer_identity=true` and see if you can reproduce the error?

**Comment by pengq-google** [comment]
After discussing with my team, we would like to make the default behavior in Terraform to always set unique_writer_identity=true. What is the best way to do this?

**Comment by shuyama1** [comment]
Changing default value in Terraform is a [breaking change](https://googlecloudplatform.github.io/magic-modules/develop/breaking-changes/#field-level-breaking-changes) that we can't release it in a minor release unfortunately. While the next major release 5.0 is targeting mid September, and we're currently accepting PRs for 5.0. More details on how to make changes for major release at https://googlecloudplatform.github.io/magic-modules/develop/make-a-breaking-change/. Please let me know if you have any questions.

File: mmv1/third_party/terraform/services/logging/resource_logging_project_sink.go [hunk 1]
```diff
@@ -36,8 +36,7 @@ func ResourceLoggingProjectSink() *schema.Resource {
 	schm.Schema["unique_writer_identity"] = &schema.Schema{
 		Type:        schema.TypeBool,
 		Optional:    true,
-		Default:     false,
-		ForceNew:    true,
+		Default:     true,
 		Description: `Whether or not to create a unique identity associated with this sink. If false (the default), then the writer_identity used is serviceAccount:cloud-logs@system.gserviceaccount.com. If true, then a unique service account is created and used for this sink. If you wish to publish logs across projects, you must set unique_writer_identity to true.`,
```
Comment (shuyama1) [inline mmv1/third_party/terraform/services/logging/resource_logging_project_sink.go:40]: may need to change the description as we change the default value.

**Review by shuyama1 — CHANGES_REQUESTED** [review]
Can you add/ modify a test to test the update functionality of the field?

**Comment by shuyama1** [comment]
rerun the build since it changed the base branch 
/gcbrun

File: mmv1/third_party/terraform/services/logging/resource_logging_project_sink_test.go [hunk 2]
```diff
@@ -294,8 +294,6 @@ resource "google_logging_project_sink" "basic" {
   project     = "%s"
   destination = "storage.googleapis.com/${google_storage_bucket.log-bucket.name}"
   filter      = "logName=\"projects/%s/logs/compute.googleapis.com%%2Factivity_log\" AND severity>=ERROR"
-
-  unique_writer_identity = false
```
Comment (shuyama1) [inline mmv1/third_party/terraform/services/logging/resource_logging_project_sink_test.go:298]: Do you still have configs that set `unique_writer_identity = false`? Just want to make sure we cover the case of setting negative value even we changed the default value in TF

**Review by shuyama1 — COMMENTED** [review]
I'd propose to leave the change of removing `forcenew` (plus a test coverage) to a separate PR open against main. Supporting in-place update is not breaking, so we can release in a minor before the major release happens. Plus, we have major release focus on breaking changes primary, so it would be easier to limit one functionality per PR for further fix or revert if needed. Sorry about the extra work it may require and thank you for working on this.

File: mmv1/third_party/terraform/services/logging/resource_logging_project_sink_test.go (hunk 2 above)
Comment (pengq-google) [reply mmv1/third_party/terraform/services/logging/resource_logging_project_sink_test.go:298]: yes, we have, for example: resource "google_logging_project_sink" "described"

**Comment by pengq-google** [comment]
> 

Just made #8844 

We just want to make sure we can catch the major release window. Thanks!

**Pull request closed without merging**